File: Src/java/cql-to-elm/src/jvmTest/java/org/cqframework/cql/cql2elm/OptionsTests.java
package org.cqframework.cql.cql2elm;

import kotlinx.io.Buffer;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.io.InputStream;

import static kotlinx.io.CoreKt.buffered;
import static kotlinx.io.JvmCoreKt.asSource;
import static kotlinx.io.Utf8Kt.readString;
import static org.junit.jupiter.api.Assertions.*;

class OptionsTests {

    @Test
    void translatorOptions() throws IOException {
        var options = CqlTranslatorOptions.defaultOptions();
        Buffer buffer = new Buffer();
        options.toSink(buffer);
        String result = readString(buffer);
        assertNotNull(result);

        InputStream input = OptionsTests.class.getResourceAsStream("options.json");
        CqlTranslatorOptions readOptions = CqlTranslatorOptions.fromSource(buffered(asSource(input)));
        assertTrue(readOptions != null);

        Buffer buffer2 = new Buffer();
        readOptions.toSink(buffer2);
        String result2 = readString(buffer2);
        assertEquals(result, result2);
    }
}

File: cql-to-elm/src/jvmTest/kotlin/org/cqframework/cql/cql2elm/OptionsTests.kt
package org.cqframework.cql.cql2elm

import java.io.IOException
import kotlinx.io.Buffer
import kotlinx.io.asSource
import kotlinx.io.buffered
import kotlinx.io.readString
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class OptionsTests {
    @Test
    @Throws(IOException::class)
    fun translatorOptions() {
        val options = CqlTranslatorOptions.defaultOptions()
        val buffer = Buffer()
        options.toSink(buffer)
        val result = buffer.readString()
        Assertions.assertNotNull(result)

        val input = OptionsTests::class.java.getResourceAsStream("options.json")
        val readOptions = CqlTranslatorOptions.fromSource(input!!.asSource().buffered())

        val buffer2 = Buffer()
        readOptions.toSink(buffer2)
        val result2 = buffer2.readString()
        Assertions.assertEquals(result, result2)
    }
}
